de.metas.procurement.base/src/main/sql/postgresql/system/50-de.metas.procurement/5455010_sys_FRESH-890_SeqNoInPMMProduct.sql

```diff
@@ -0,0 +1,93 @@
+;
+
+-- 03.01.2017 15:48
+-- URL zum Konzept
+INSERT INTO AD_Field_Trl (AD_Language,AD_Field_ID, Description,Help,Name, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy) SELECT l.AD_Language,t.AD_Field_ID, t.Description,t.Help,t.Name, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy FROM AD_Language l, AD_Field t WHERE l.IsActive='Y' AND l.IsSystemLanguage='Y' AND l.IsBaseLanguage='N' AND t.AD_Field_ID=557492 AND NOT EXISTS (SELECT * FROM AD_Field_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Field_ID=t.AD_Field_ID)
+;
+
+-- 03.01.2017 16:06
+-- URL zum Konzept
+UPDATE AD_Column SET DefaultValue='', IsMandatory='N',Updated=TO_TIMESTAMP('2017-01-03 16:06:39','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=556027
+;
+
+-- 03.01.2017 16:07
+-- URL zum Konzept
+UPDATE AD_Column SET DefaultValue='@SQL=SELECT COALESCE(MAX(SeqNo),0)+10 AS DefaultValue FROM PMM_Product WHERE M_Product_ID=@M_Product_ID@ AND M_HU_PI_Item_Product_ID = @M_HU_PI_Item_Product_ID@ AND (C_BPartner_ID = @C_BPartner_ID@ OR C_BPartner_ID IS NULL)
+',Updated=TO_TIMESTAMP('2017-01-03 16:07:01','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=556027
+;
+
+-- 03.01.2017 16:24
+-- URL zum Konzept
+UPDATE AD_Index_Column SET AD_Column_ID=556027,Updated=TO_TIMESTAMP('2017-01-03 16:24:15','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Index_Column_ID=540751
+;
+
+-- 03.01.2017 16:24
+-- URL zum Konzept
+UPDATE AD_Index_Column SET ColumnSQL='COALESCE(SeqNo,0)',Updated=TO_TIMESTAMP('2017-01-03 16:24:39','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Index_Column_ID=540751
+;
+
+-- 03.01.2017 16:25
+-- URL zum Konzept
+UPDATE AD_Column SET DefaultValue='@SQL=SELECT COALESCE(MAX(SeqNo),0)+10 AS DefaultValue FROM PMM_Product WHERE M_Product_ID=@M_Product_ID@ AND M_HU_PI_Item_Product_ID = @M_HU_PI_Item_Product_ID@ AND C_BPartner_ID = @C_BPartner_ID/-1@ 
+',Updated=TO_TIMESTAMP('2017-01-03 16:25:39','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=556027
+;
+
+-- 03.01.2017 16:28
+-- URL zum Konzept
+UPDATE AD_Column SET AD_Reference_ID=11,Updated=TO_TIMESTAMP('2017-01-03 16:28:23','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=556027
+;
+
+-- 03.01.2017 16:29
+-- URL zum Konzept
+ALTER TABLE public.PMM_Product ADD SeqNo NUMERIC(10) DEFAULT NULL 
+;
+
+DROP INDEX public.pmm_product_uc;-- 03.01.2017 16:45
+
+
+
+
+UPDATE PMM_Product SET SeqNo = 10 where M_AttributeSetInstance_ID IS  NULL;
+
+UPDATE PMM_Product SET SeqNo = 20 where M_AttributeSetInstance_ID IS NOT NULL;
+
+
+--- !!!!!!!!!!!!!!!!!! IF THERE ARE MORE THAN 2 PMM_PRODUCT ENTRIES FOR THE SAME PRODUCT, M_HU_PI_ITEM_PRODUCT AND PARTNER, PLEASE UPDATE THE SEQNO MANUALLY HERE!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
Phew, that's a bit problematic...
we usually only have a 15 minutes rollout window. If this SQL fails because some user created some `PMM_Prodcut`s, we have a problem and might not be able to fix things in time..

It should be possible to solve this with an update that's getting the `SeqNo` valued from a select with a *window function*
..though i don't know how exactly, but am pretty sure it wou,ld work... maybe @metas-lc can help?